dart/lib/src/scope.dart

```diff
@@ -83,39 +80,42 @@ class Scope {
 
   final SentryOptions _options;
 
-  Scope(this._options) {
-    if (_options == null) {
-      throw ArgumentError('SentryOptions is required');
-    }
-  }
+  Scope(this._options);
 
   /// Adds a breadcrumb to the breadcrumbs queue
   void addBreadcrumb(Breadcrumb breadcrumb, {dynamic hint}) {
-    assert(breadcrumb != null, "Breadcrumb can't be null");
-
     // bail out if maxBreadcrumbs is zero
     if (_options.maxBreadcrumbs == 0) {
       return;
     }
 
     // run before breadcrumb callback if set
-    if (_options.beforeBreadcrumb != null) {
-      breadcrumb = _options.beforeBreadcrumb(breadcrumb, hint: hint);
-
-      if (breadcrumb == null) {
-        _options.logger(
-            SentryLevel.info, 'Breadcrumb was dropped by beforeBreadcrumb');
-        return;
+    if (_options.beforeBreadcrumb == null) {
+      // remove first item if list if full
+      if (_breadcrumbs.length >= _options.maxBreadcrumbs &&
+          _breadcrumbs.isNotEmpty) {
+        _breadcrumbs.removeFirst();
```

Comment:
we duplicated this logic and previously we did it only once, how this change make it better?

Reply:
It didn't, I'll change it